config/deprecation.go

```diff
@@ -0,0 +1,362 @@
+	infos := make(map[string][]pluginDeprecationInfo)
+
+	infos["inputs"] = make([]pluginDeprecationInfo, 0)
+	for name, creator := range inputs.Inputs {
+		if len(inFilter) > 0 && !sliceContains(name, inFilter) {
+			continue
+		}
+
+		plugin := creator()
+		info := c.collectDeprecationInfo("inputs", name, plugin, true)
+
+		if info.Level != None || len(info.Options) > 0 {
+			infos["inputs"] = append(infos["inputs"], info)
+		}
+	}
+
+	infos["outputs"] = make([]pluginDeprecationInfo, 0)
+	for name, creator := range outputs.Outputs {
+		if len(outFilter) > 0 && !sliceContains(name, outFilter) {
+			continue
+		}
+
+		plugin := creator()
+		info := c.collectDeprecationInfo("outputs", name, plugin, true)
+
+		if info.Level != None || len(info.Options) > 0 {
+			infos["outputs"] = append(infos["outputs"], info)
+		}
+	}
+
+	infos["processors"] = make([]pluginDeprecationInfo, 0)
+	for name, creator := range processors.Processors {
+		if len(procFilter) > 0 && !sliceContains(name, procFilter) {
+			continue
+		}
+
+		plugin := creator()
+		info := c.collectDeprecationInfo("processors", name, plugin, true)
+
+		if info.Level != None || len(info.Options) > 0 {
+			infos["processors"] = append(infos["processors"], info)
+		}
+	}
+
+	infos["aggregators"] = make([]pluginDeprecationInfo, 0)
+	for name, creator := range aggregators.Aggregators {
+		if len(aggFilter) > 0 && !sliceContains(name, aggFilter) {
+			continue
+		}
+
+		plugin := creator()
+		info := c.collectDeprecationInfo("aggregators", name, plugin, true)
+
+		if info.Level != None || len(info.Options) > 0 {
+			infos["aggregators"] = append(infos["aggregators"], info)
+		}
+	}
+
+	return infos
+}
+
+func (c *Config) PrintDeprecationList(plugins []pluginDeprecationInfo) {
+	sort.Slice(plugins, func(i, j int) bool { return plugins[i].Name < plugins[j].Name })
+
+	for _, plugin := range plugins {
+		switch plugin.Level {
+		case Warn, Error:
+			_, _ = fmt.Printf(
+				"  %-40s %-5s since %-5s removal in %-5s %s\n",
+				plugin.Name, plugin.Level, plugin.info.Since, plugin.info.RemovalIn, plugin.info.Notice,
+			)
+		}
+
+		if len(plugin.Options) < 1 {
+			continue
+		}
+		sort.Slice(plugin.Options, func(i, j int) bool { return plugin.Options[i].Name < plugin.Options[j].Name })
+		for _, option := range plugin.Options {
+			_, _ = fmt.Printf(
+				"  %-40s %-5s since %-5s removal in %-5s %s\n",
+				plugin.Name+"/"+option.Name, option.Level, option.info.Since, option.info.RemovalIn, option.info.Notice,
+			)
+		}
+	}
+}
+
+func printHistoricPluginDeprecationNotice(category, name string, info telegraf.DeprecationInfo) {
+	prefix := "E! " + color.RedString("DeprecationError")
+	log.Printf(
+		"%s: Plugin %q deprecated since version %s and removed: %s",
+		prefix, category+"."+name, info.Since, info.Notice,
+	)
+}
+
+func printPluginDeprecationNotice(prefix, name string, info telegraf.DeprecationInfo) {
+}
```

Comment:
Empty function? Intentional?

Reply:
Ups. Definitively not intended! Will fix.